http-server-netty/src/test/groovy/io/micronaut/http/server/netty/binding/HttpResponseSpec.groovy

```diff
@@ -232,6 +229,7 @@ class HttpResponseSpec extends AbstractMicronautSpec {
         server.close()
     }
 
+    @Ignore // why wouldn't keep-alive be set?
     void "test keep alive connection header is not set by default for > 499 response"() {
```

Comment:
For my own education, are we getting 'keep-alive' now as the underlying netty server has changed to return 'keep-alive' in the error response? Or are we getting it now because or the client changes and the fact we're now using a pool by default?

I've done some looking around and it seems historically it made some sense to 'close' on errors for a server serving html to people (users are likely to just go elsewhere), but I don't think there's an issue here.

I was just interested in what caused this change in behaviour.

If we stick with keep-alive, and it's being set by the server, I guess we can remove `micronaut.server.netty.keepAliveOnServerError`?  Or do we need config to switch back to use the old close header?

Reply:
> For my own education, are we getting 'keep-alive' now as the underlying netty server has changed to return 'keep-alive' in the error response? Or are we getting it now because or the client changes and the fact we're now using a pool by default?

Yea, it's the latter. The server never forced `close`, but because before this pr the client requested using `Connection: close`, so the server replied with that as well. Now that the pool is enabled, the server is perfectly happy to stay `keep-alive`. I guess this test case just didn't test for the feature properly.

> If we stick with keep-alive, and it's being set by the server, I guess we can remove micronaut.server.netty.keepAliveOnServerError? Or do we need config to switch back to use the old close header?

I didn't know about that setting, I will take a look at why it doesn't work as intended.

Reply:
> I didn't know about that setting, I will take a look at why it doesn't work as intended.

I only saw it when I was trying the Ignored test locally 🕵️‍♀️

Reply:
the logic was a bit weird, ive fixed it and enabled the setting by default.